### PR TITLE
Add menu tab widget for foreign ships

### PR DESCRIPTION
**Pull request by hepfnepf**
I added a menu tab widget for non player ships: ~ShipAlienMenuTabWidget~`ShipForeignMenuTabWidget`. It displays the owner's correct flag and the ship name. The normal `ShipMenuTabWidget` now also displays the name of the ship.

### CODE REVIEW

**Comment by LinuxDonald** [comment]
Thank you @artism90 will review it.

File: Assets/Player/PlayerHUD.gd [hunk 1]
```diff
@@ -12,6 +12,7 @@ enum UIContext {
 	NONE,
 	TROOP,
 	SHIP,
+	SHIP_ALIEN,
```
Comment (pingwindev) [inline Assets/Player/PlayerHUD.gd]:
```suggestion
	SHIP_FOREIGN,
```
"Alien" sounds weird in this context. Let it be "Foreign".

File: Assets/Player/PlayerHUD.gd [hunk 2]
```diff
@@ -135,7 +136,14 @@ func _on_PlayerCamera_selected(selected_entities: Array) -> void:
...
-			new_context = UIContext.SHIP
+			if entity.faction == Global.Game.player.faction:
+				new_context = UIContext.SHIP
+			else:
+				new_context = UIContext.SHIP_ALIEN
+			context_data = {
+				"FactionIndicator":Global.FACTION_FLAGS[entity.faction],
+				"Caption":entity.unit_name
+				}
 			break
```
Comment (pingwindev) [inline Assets/Player/PlayerHUD.gd]:
```suggestion
				new_context = UIContext.SHIP_FOREIGN
			
			context_data = {
				"FactionIndicator": Global.FACTION_FLAGS[entity.faction],
				"Caption": entity.unit_name,
			}
			
			break
```
Update to `UIContext.SHIP_FOREIGN` and formatting issues.

File: Assets/UI/Scripts/ShipAlienMenuTabWidget.gd [hunk 3]
```diff
@@ -0,0 +1,23 @@
+tool
+extends TabWidget
+class_name ShipAlienMenuTabWidget
```
Comment (pingwindev) [inline Assets/UI/Scripts/ShipAlienMenuTabWidget.gd]:
```suggestion
class_name ShipForeignMenuTabWidget
```

File: Assets/UI/Scripts/ShipAlienMenuTabWidget.gd [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+
+func update_data(context_data: Dictionary) -> void:
+	for data in context_data:
+		prints("data:", data) # TownName
+		var node = find_node(data)
+
+		if node is Label:
+			node.text = context_data[data]
+	
+		if (data == "FactionIndicator"):
```
Comment (pingwindev) [inline Assets/UI/Scripts/ShipAlienMenuTabWidget.gd]:
```suggestion
		if data == "FactionIndicator":
```
Redundant parentheses.

**Review by pingwindev — CHANGES_REQUESTED** [review]
See review notes. Also rename the tab widget scene to `ShipForeignMenuTabWidget`.

Furthermore, always save files with `Trim Trailing Whitespace On Save` enabled:
Screenshot 2022-03-19 132707

**Review by pingwindev — CHANGES_REQUESTED** [review]
The node name of the embedded tab widget inside `PlayerHUD.tscn` still must be renamed.

**Note:** Do not add a new commit for this single fix, but amend it to the previous one. You can do it by `git commit --amend` or with any advanced git GUI application (e.g. GitKraken).

image

**Comment by hepfnepf** [comment]
Sorry for not amending. Read that part later.

**Pull request merged**